sql/testdata/snapshot

```diff
@@ -0,0 +1,58 @@
+# Run a simple shell session in which a reader conflicts with an open txn in
+# SNAPSHOT isolation by means of a read, which is resolved by a Push. Prevents
+# regression of #2861.
```

Comment:
I noticed in go that they have test cases named after the issue they reproduce (e.g. https://github.com/golang/go/blob/master/test/fixedbugs/issue11326.go). Seems like a nice pattern to follow: `s,testdata/snapshot,testdata,issue2861,g`